Changelog

```diff
@@ -1,4 +1,40 @@
-Version 2.21.4, 
+Version 2.22, TDB
+
+  Features:
+  * Add automatic offline refill for Offline OTP tokens (#839)
+  * Return realm and resolver of the user and allow mapping
+    group membership to the RADIUS protocol (#896)
+  * Add new tokenkind (hardware, software, virtual) for all tokens (#828)
+  * Support Vasco tokens via Import and via Web Enrollment (#904, #903, #891)
+  * Add arbitrary tokeninfo field to authorization policy (#873)
+  * New SMPP SMS provider (#878)
+
+  Enhancements:
+  * Enhance the statistics possibilities in WebUI (#950)
+  * Allow reencryption of the database by importing PSKC to
+    a new database (#940)
+  * Allow token janitor to export "PW" token type to PSKC (#942)
+  * Also export and import the counter values of HOTP/TOTP to PSKC (#943)
+  * SMS token can dynamically read phone number from user source (#932)
+  * Email token can dynamically read email address from user source (#932)
+  * Add policy to ignore the validity of a U2F attestation certificate (#926)
+  * Improve the speed of the LinOTP migration script to cope with tens of
+    thousands of tokens (#914)
+  * pi-manage can create API tokens with a chosen validity time (#931)
+  * Allow user to set token description for HOTP and TOTP tokens 
+    during enrollment (#928)
+  * Add timeout to SMTP server configuration (#919)
+  * Allow complex email templates for email tokens (#684)
+  * LDAP resolver now supports arbitrary multivalue attributes (#881)
```

Comment:
Ah, this misses #971 and #929 (though the latter may also be a fix).

Reply:
I will add #971 as enhancement
I will add #929 to fixes.